### ISSUE TITLE
documentation of supported onnx operations

### ISSUE THREAD

**Issue by noahbadoa** [opencv/opencv#22301]
I couldn't find any documentation after a module with "RandomUniform" which is in the first onnx opset failed to compile. What is and isn't supported should probably be somewhere other than source code.

**Comment by zihaomu** [comment]
Can you attach your ONNX model to the issue?

**Comment by noahbadoa** [comment]
It's a modified unet for image to image generation. I'm pretty sure the problem is with torch.rand_like and torch.randn.

"""
class DoubleConv(nn.Module):
    def __init__(self, in_channels, out_channels, mid_channels=None):
        super().__init__()
        if not mid_channels:
            mid_channels = out_channels
        self.double_conv = nn.Sequential(
            nn.Conv2d(in_channels, mid_channels, kernel_size=3, padding=1, bias=False),
            nn.BatchNorm2d(mid_channels),
            nn.ReLU(),
            nn.Conv2d(mid_channels, out_channels, kernel_size=3, padding=1, bias=False),
            nn.BatchNorm2d(out_channels),
            nn.ReLU()
        )

    def forward(self, x):
        return self.double_conv(x)

class Down(nn.Module):
    def __init__(self, in_channels, out_channels):
        super().__init__()
        self.maxpool_conv = nn.Sequential(
            nn.MaxPool2d(2),
            DoubleConv(in_channels, out_channels)
        )

    def forward(self, x):
        return self.maxpool_conv(x)

class Noised_Up(nn.Module):
    def __init__(self, in_channels, out_channels, bilinear=True):
        super().__init__()

        # if bilinear, use the normal convolutions to reduce the number of channels
        if bilinear:
            self.up = nn.Upsample(scale_factor=2, mode='bilinear', align_corners=True)
            self.conv = DoubleConv(in_channels + 32, out_channels, in_channels // 2)
        else:
            self.up = nn.ConvTranspose2d(in_channels, in_channels // 2, kernel_size=2, stride=2)
            self.conv = DoubleConv(in_channels + 32, out_channels)
        
        self.scale = nn.Parameter(torch.rand(2))

    def forward(self, x1, x2):
        x1 = self.up(x1 + torch.rand_like(x1) * self.scale[0])
        # input is CHW
        diffY = x2.size()[2] - x1.size()[2]
        diffX = x2.size()[3] - x1.size()[3]

        x1 = F.pad(x1, [diffX // 2, diffX - diffX // 2,
                        diffY // 2, diffY - diffY // 2])
        x = torch.cat([x2, x1, torch.randn(x1.size(0), 32, x1.size(2), x1.size(3), device = x1.device)], dim=1)
        x = self.conv(x) 
        return x+ torch.rand_like(x) * self.scale[1]


class UNet(nn.Module):
    def __init__(self, n_channels, n_out, bilinear = True, start = 72):
        super(UNet, self).__init__()
        self.n_channels = n_channels
        self.n_out = n_out
        self.bilinear = bilinear

        self.inc = DoubleConv(n_channels, start)
        self.down2 = Down(start, start * 2)
        self.down3 = Down(start * 2, start * 4)
        factor = 2 if bilinear else 1
        self.down4 = Down(start * 4, start * 8 // factor)
        self.up1 = Noised_Up(start * 8, start * 4 // factor, bilinear)
        self.up2 = Noised_Up(start * 4, start * 2 // factor, bilinear)
        self.up3 = Noised_Up(start * 2, start // factor, bilinear)
        self.outc = nn.Sequential(
            nn.Conv2d(start // factor, start // factor, kernel_size=1),
            nn.GELU(),
            nn.Conv2d(start // factor, n_out, kernel_size=1),
        )

        factor = 1 if bilinear else 2

    def forward(self, x):
        x1 = self.inc(x)
        x2 = self.down2(x1)
        x3 = self.down3(x2)
        x4 = self.down4(x3)
        x = self.up1(x4, x3)
        x = self.up2(x, x2)
        x = self.up3(x, x1)
        logits = self.outc(x)
        return torch.tanh(logits)

"""

**Comment by zihaomu** [comment]
Can you attach the exported ONNX file to the issue? Raw pytorch code can't help.
BTW, you can find the supported Op at [this wiki page](https://github.com/opencv/opencv/wiki/Deep-Learning-in-OpenCV).

**Comment by noahbadoa** [comment]
[test (1).zip](https://github.com/opencv/opencv/files/9215164/test.1.zip)

**Comment by zihaomu** [comment]
Thanks @noahbadoa, I found that there are some layer in model that we don't support, like `RandomUniformLike` and `ConstantOfShape`. And I also found the weight and bias of `Conv layer` in your model are variable, that is no good for OpenCV.

**Comment by asmorkalov** [comment]
Answering original question: https://github.com/opencv/opencv/wiki/Deep-Learning-in-OpenCV. The list is updated manually and may be incomplete. Please use models dignostic tool to check model support in future: https://github.com/opencv/opencv/wiki/OpenCV-Debugging-Facilities#dnn
